core/src/test/java/org/elasticsearch/indices/IndicesServiceTests.java

```diff
@@ -187,4 +199,54 @@ public void testPendingTasks() throws Exception {
         assertAcked(client().admin().indices().prepareOpen("test"));
 
     }
+
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/17695")
+    public void testDeletingSameIndexDirectoryFromConcurrentProcesses() throws Exception {
```

Comment:
The reason I put this test in was to simulate the behavior of Point No. 2 in #17695 and when that issue gets resolved, we can remove the `AwaitsFix` annotation.  What I think you're suggesting is, the fix for the issue of nodes concurrently deleting the same index directory may not involve a fix at such a low level, so we should just wait to see how we tackle it before writing such a low level test for it?